File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    bool check(stack<char> &st,char ch){
        if(st.top() == 'A' && ch == 'B')
            return true;
        if(st.top() == 'C' && ch == 'D')
            return true;
        return false;
    }
    
    int minLength(string s) {
        stack<char> st;
        int i=0;
        
        while(i<s.size()){
            while(i<s.size() && !st.empty() && check(st,s[i])){
                st.pop();
                i++;
            }
            if(i == s.size())
                break;
            st.push(s[i]);
            i++;
        }
        return st.size();
    }
};

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.java
class Solution {
    public boolean check(Stack<Character> st,char ch){
        if(st.peek() == 'A' && ch == 'B')
            return true;
        if(st.peek() == 'C' && ch == 'D')
            return true;
        return false;
    }
    
    public int minLength(String s) {
        Stack<Character> st = new Stack<>();
        
        int i=0;
        while(i<s.length()){
            while(i<s.length() && !st.isEmpty() && check(st,s.charAt(i))){
                st.pop();
                i++;
            }
            if(i == s.length())
                break;
            st.push(s.charAt(i));
            i++;
        }
        return st.size();
    }
}